Add TORCH_LOGS_OUT to direct TORCH_LOGS output

oulgen opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117005
* #116894

Twice now, while I was debugging accuracy bugs, I get dynamo logs that are 100k lines long and it is impossible to read them on the terminal. Lets add an option to write them to a file.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

oulgen committed:
Add TORCH_LOGS_OUT to direct TORCH_LOGS output

[ghstack-poisoned]

ezyang reviewed (approved):
Sure. You might want to add a test, our current logging tests should have the scaffolding you need

oulgen commented:
> Sure. You might want to add a test, our current logging tests should have the scaffolding you need

I didn't even know we had logging tests, i couldnt find them via simple grep. Can you point me to them?

oulgen committed:
Update on "Add TORCH_LOGS_OUT to direct TORCH_LOGS output"


Twice now, while I was debugging accuracy bugs, I get dynamo logs that are 100k lines long and it is impossible to read them on the terminal. Lets add an option to write them to a file.


[ghstack-poisoned]

oulgen commented:
@pytorchbot merge
